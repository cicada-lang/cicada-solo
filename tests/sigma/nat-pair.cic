one = add1(zero)
two = add1(one)
three = add1(two)
four = add1(three)

[_: Nat | Nat]
Pair(Nat, Nat)

one_two: [_: Nat | Nat] = cons(one, two)
cdr(one_two)

[_: Nat, _: Nat | Nat]
Pair(Nat, Pair(Nat, Nat))

one_two_three: [_: Nat, _: Nat | Nat] = cons(one, cons(two, three))
cdr(cdr(one_two_three))

[_: Nat, _: Nat, _: Nat | Nat]
Pair(Nat, Pair(Nat, Pair(Nat, Nat)))

one_two_three_four: [_: Nat, _: Nat, _: Nat | Nat] = cons(one, cons(two, cons(three, four)))
cdr(cdr(cdr(one_two_three_four)))
