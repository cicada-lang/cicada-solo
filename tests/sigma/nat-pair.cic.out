[_: Nat | Nat]: Type
[_: Nat | Nat]: Type
2: Nat
[_: Nat, _1: Nat | Nat]: Type
[_: Nat, _1: Nat | Nat]: Type
3: Nat
[_: Nat, _1: Nat, _2: Nat | Nat]: Type
[_: Nat, _1: Nat, _2: Nat | Nat]: Type
4: Nat